src/main/java/ro/siit/trips/controllers/TripController.java

```diff
@@ -0,0 +1,190 @@
+                    break;
+                }
+            }
+        }
+        if (currentTrip == null) {
+            currentTrip = loggedPerson.getTrips().iterator().next();
+
+        }
+        model.addObject("trips", loggedPerson.getTrips());
+        model.addObject("selectedId", currentTrip.getId());
+        model.addObject("currentTrip", currentTrip);
+        return model;
+
+    }
+
+    @RequestMapping(value = "/delete", method = RequestMethod.POST)
+    public ModelAndView delete(@ModelAttribute Trip trip, @RequestParam(name = "id", required = false) Long id) {
+        ModelAndView model = new ModelAndView("tripTemplate");
+        tripRepository.deleteById(trip.getId());
+        Person loggedPerson = personRepository.findByUserName(SecurityContextHolder.getContext().getAuthentication().getName());
+        System.out.println(loggedPerson.getTrips());
+        Trip currentTrip = null;
+        if (id != null) {
+            for (Trip trip2 : loggedPerson.getTrips()) {
+                if (trip.getId().equals(id)) {
+                    currentTrip = trip;
+                    System.out.println("Found:" + trip);
+                    break;
+                }
+            }
+        }
+        if (currentTrip == null) {
+            currentTrip = loggedPerson.getTrips().iterator().next();
+
+        }
+
+        model.addObject("trips", loggedPerson.getTrips());
+        model.addObject("selectedId", currentTrip.getId());
+        model.addObject("currentTrip", currentTrip);
+        return model;
+    }
+
+
+    @RequestMapping(value = "/tripToUpdate", method = RequestMethod.POST)
+    public ModelAndView findingTripForEdit(@ModelAttribute Trip trip) {
+        ModelAndView model = new ModelAndView("/updateTrip");
+        Trip modifiedTrip = tripRepository.findById(trip.getId()).get();
+        tripRepository.save(modifiedTrip);
+        model.addObject("trip", modifiedTrip);
+        return model;
+    }
+
+    @RequestMapping(value = "/updatedTrip", method = RequestMethod.POST)
+    public ModelAndView editingTrip(@ModelAttribute Trip trip, @RequestParam(name = "id", required = false) Long id,
+                                    @RequestParam("file") MultipartFile file,
+                                    @RequestParam("file2") MultipartFile file2) {
+        ModelAndView model = new ModelAndView("tripTemplate");
+        Person loggedPerson = personRepository.findByUserName(SecurityContextHolder.getContext().getAuthentication().getName());
+        trip.setPerson(loggedPerson);
+        Trip modifiedTrip = tripRepository.findById(trip.getId()).get();
+        modifiedTrip.setTripName(trip.getTripName());
+
+        modifiedTrip.setImpressions(trip.getImpressions());
+        modifiedTrip.setStartDate(trip.getStartDate());
+        modifiedTrip.setEndDate(trip.getEndDate());
+        modifiedTrip.setLocation(trip.getLocation());
+        if (file.isEmpty()) {
+            modifiedTrip.setPhoto1(modifiedTrip.getPhoto1());

```

Comment:
this is useless.. you should reverse the if and just do the else branch:
if(!file.isEmpty()) {
  modifiedTrip.setPhoto1(UUID.randomUUID().toString());
            fileService.store(modifiedTrip.getPhoto1(), file);
}